src/ml/neural_net/image_augmentation.cpp

```diff
@@ -198,8 +204,15 @@ image_augmenter::result float_array_image_augmenter::prepare_images(
   float_array_result augmented_data = prepare_augmented_images(input_to_tf_aug);
 
   // Convert augmented_data to the data structure needed
-  res.image_batch = augmented_data.images;
-  for (size_t a = 0; a < n; a++) {
+  std::vector<float> result_array(n * h * w * c);
+  size_t image_size = augmented_data.images.size();
+  const float* start_address = augmented_data.images.data();
+  const float* end_address = (float*)(start_address + image_size);
```

Comment:
The `float *` should be unnecessary 

For future reference, don’t use C-style casts in C++ code anyway

Reply:
Good learn ;)

Reply:
besides, you don't need to cast type for pointer arithmetic.